kiwi/package_manager/yum.py

```diff
@@ -173,6 +173,9 @@ def process_delete_requests(self, force=False):
         """
         Process package delete requests (chroot)
 
+        Note: yum erase does not delete unneeded dependencies it only
+        makes sure no dependencies are broken.
```

Comment:
This can actually be done with `yum autoremove`

Cf. https://www.mankier.com/8/yum-deprecated#autoremove

Reply:
The version of yum I have here does not provide an autoremove command, I only see erase which was the reason I used this command in the initial implementation. I we can asure yum (like on rhel) has it I think it would be good to move to autoremove. If only yum-deprecated provides the command I think we should stick to the current implementation

Reply:
@schaefi, I also used erase because of this, I do not see autoremove in `yum --help` in the version I have and I see the same in Centos 7.0, however I have just tested the image we have in kiwi-descriptions using the `yum autoremove` and it worked fine as @Conan-Kudo described. I guess there is lack of documentation in `--help` option. I haven't checked man pages as it uses the `excludedocs` flag.

I am about to commit a change to make use of `autoremove`.

Reply:
ah ok, if a build works I think we are good to go

Reply:
@davidcassany It's in RHEL 7, it was introduced in https://github.com/rpm-software-management/yum/commit/4ad8960e12a19cbd2f6e3bb039c7b65015f59e19

Prior to this, `yum --setopt=clean_requirements_on_remove=True remove` also worked.

Reply:
ok, thanks, so if no further objections I would merge it

Reply:
Go for it!